<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EdgeProcessingWidget</class>
 <widget class="QWidget" name="EdgeProcessingWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edge Processing</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="labelTitle">
     <property name="text">
      <string>Edge Detection Algorithms</string>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTableWidget" name="tableAlgorithms">
     <property name="selectionMode">
      <enum>QAbstractItemView::SingleSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <column>
      <property name="text">
       <string>Algorithm</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Description</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupParameters">
     <property name="title">
      <string>Parameters</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="labelParam1">
        <property name="text">
         <string>Low Threshold:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QSlider" name="sliderParam1">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinParam1">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>50</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelParam2">
        <property name="text">
         <string>High Threshold:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QSlider" name="sliderParam2">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>150</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinParam2">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="value">
           <number>150</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelParam3">
        <property name="text">
         <string>Aperture Size:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="comboParam3">
        <item>
         <property name="text">
          <string>3</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>5</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>7</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="labelParam4">
        <property name="text">
         <string>Kernel Size:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="spinParam4">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>31</number>
        </property>
        <property name="singleStep">
         <number>2</number>
        </property>
        <property name="value">
         <number>3</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="labelParam5">
        <property name="text">
         <string>Scale:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinParam5">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0.010000000000000</double>
        </property>
        <property name="maximum">
         <double>10.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="labelParam6">
        <property name="text">
         <string>Delta:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QDoubleSpinBox" name="doubleSpinParam6">
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="maximum">
         <double>255.000000000000000</double>
        </property>
        <property name="value">
         <double>0.000000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="checkL2Gradient">
     <property name="text">
      <string>Use L2 Gradient</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="btnApply">
       <property name="text">
        <string>Apply</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnReset">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QPushButton" name="btnSaveParams">
       <property name="text">
        <string>Save Parameters</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnLoadParams">
       <property name="text">
        <string>Load Parameters</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>sliderParam1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinParam1</receiver>
   <slot>setValue(int)</slot>
  </connection>
  <connection>
   <sender>spinParam1</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sliderParam1</receiver>
   <slot>setValue(int)</slot>
  </connection>
  <connection>
   <sender>sliderParam2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinParam2</receiver>
   <slot>setValue(int)</slot>
  </connection>
  <connection>
   <sender>spinParam2</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sliderParam2</receiver>
   <slot>setValue(int)</slot>
  </connection>
 </connections>
</ui>
